php_phongo.c

```diff
@@ -527,6 +538,34 @@ static bool process_read_preference(zval *option, bson_t *mongoc_opts, zval **zr
 	return true;
 }
 
+static bool process_session(zval *option, bson_t *mongoc_opts, zval **zsession, mongoc_client_t *client TSRMLS_DC)
+{
+	const mongoc_client_session_t *client_session;
+
+	if (Z_TYPE_P(option) != IS_OBJECT || !instanceof_function(Z_OBJCE_P(option), php_phongo_session_ce TSRMLS_CC)) {
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected \"session\" option to be %s, %s given", ZSTR_VAL(php_phongo_session_ce->name), zend_get_type_by_const(Z_TYPE_P(option)));
+		return false;
+	}
+
+	client_session = Z_SESSION_OBJ_P(option)->client_session;
+
+	if (client != mongoc_client_session_get_client(client_session)) {
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Cannot use Session started from a different Manager");
+		return false;
+	}
+
+	if (!mongoc_client_session_append(Z_SESSION_OBJ_P(option)->client_session, mongoc_opts, NULL)) {
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Error appending \"session\" option");
```

Comment:
Is this likely to happen? If so, we should improve our error message to suggest how users can fix it.

Reply:
Very unlikely. BSON append for primitive types only fails if the [buffer cannot grow](https://github.com/mongodb/mongo-c-driver/blob/6fa2c35119672c8ce7c0ac9a926fc33ecc3b7ae1/src/libbson/src/bson/bson.c#L325) or the [document length exceeds a 32-bit integer range](https://github.com/mongodb/mongo-c-driver/blob/6fa2c35119672c8ce7c0ac9a926fc33ecc3b7ae1/src/libbson/src/bson/bson.c#L394). Similar to other places where we check for failure (e.g. appending "serverId"), I'm just being a good libmongoc citizen here.